optuna/study.py

```diff
@@ -751,8 +751,12 @@ def _run_trial(
 
         return trial
 
-    def _log_completed_trial(self, trial, result):
-        # type: (trial_module.Trial, float) -> None
+    def _log_completed_trial(self, trial: trial_module.Trial, result: float) -> None:
+
+        if logging.get_verbosity() > logging.INFO:
+            return
+        if not (logging._is_enabled_default_handler() or logging._is_enabled_propagation()):
+            return
```

Comment:
Thank you for your careful review. The progress bar was completely out of my mind.
Although the progress bar works with the following `optuna.logging._is_enabled_default_hander`, I think checking the handler/propagation settings makes logic too complicated. I'll only check the log level to skip the storage access, and remove the `optuna.logging._is_enabled_default_hander()` and `optuna.logging._is_enabled_propagation()`.

I think it is reasonable because the [FAQ](https://optuna.readthedocs.io/en/stable/faq.html#how-to-suppress-log-messages-of-optuna) suggests `optuna.logging.set_verbosity` to suppress the log messages.

```
def _is_enabled_default_handler() -> bool:
    _configure_library_root_logger()

    assert _default_handler is not None
    return len(_get_library_root_logger().handlers) > 0
```

Reply:
I'm also for keeping it simple in this PR. It should be enough for the majority of users.